### DISCUSSION TITLE
Request quality in /1.0/content/%{asin}/licenserequest

### DISCUSSION THREAD

**Discussion by Mbucari** [mkb79/Audible#54] [Q&A] [answered]
That api endpoint purports to support 4 levels of quality: High, Normal, Extreme, Low.  I recently verified this by submitting a request for "medium" and getting a response that enumerated those four options as the only valid types.

However, the server only appears to serve up 2 different qualities. Requests for High and Extreme return the same file, and requests for Normal and Low return the same file.  This makes some amount of sense, because in the official Audible app there is only the option for High or Low.  However, many books have 4 different aax qualities listed in available_codecs. Using the old aax workaround method, you could specify any of these four qualities and get 4 different aax files. But with /1.0/content/%{asin}/licenserequest, I am only ever able to get 2 different files.

Do you know if there's a workaround to get the other levels of quality with this api endpoint?

**Comment by mkb79** [comment, marked as answer]
You are right, the licenserequest endpoint supports theoretical 4 quality levels, but only 2 where served. `audible-cli` take this into account. I doesn't know any workaround for this endpoint yet.
